internal/devpkg/package.go

```diff
@@ -284,8 +284,25 @@ func (p *Package) NormalizedPackageAttributePath() (string, error) {
 	return p.normalizedPackageAttributePathCache, nil
 }
 
+// search attempts to find the package in the local nixpkgs. It may be an expensive
+// call, if it is has not been cached.
+func (p *Package) search(query string) (map[string]*nix.Info, error) {
```

Comment:
This API is really weird. Having the package as a receiver and also requiring a query. I would suggest a different API, but I'm not sure we need this function at all. See comments below

Reply:
It's weird because you are searching for the query (which is determined by the package), but tangentially using the package information to determine how the search will happen as well.

It's a bit like

```golang
mysql := devpkg.New("mysql")
mysql.Search("mysql")
```

* Search should not have a receiver OR should not have a query param.

I'm not sure renaming it addresses the concern.

An alternative:

```golang
func (p *Package) lookupNixInfo() (map[string]*nix.Info, error) {
        var query string
	if p.IsDevboxPackage() {
		if p.isVersioned() {
			entry, err := p.lockfile.Resolve(p.Raw)
			if err != nil {
				return "", err
			}
			query = entry.Resolved
		} else {
			query = p.lockfile.LegacyNixpkgsPath(p.String())
		}
	} else {
		query = p.String()
	}
        return nix.Search(query)
}
```